app/controllers/idv/image_upload_controller.rb

```diff
@@ -0,0 +1,73 @@
+module Idv
+  class ImageUploadController < ApplicationController
+    include IdvSession
+
+    def upload
+      validation_error = validate_request(request)
+      response = validation_error || upload_and_check_images
+      render json: response
+    end
+
+    private
+
+    def upload_and_check_images
+      doc_response = client.post_images(front_image: @front_image,
+                                        back_image: @back_image,
+                                        selfie_image: @selfie_image,
+                                        liveness_checking_enabled: liveness_checking_enabled?)
+      return error_json(doc_response.errors.first) unless doc_response.success?
+      upload_info = {
+        documents: doc_response.to_h,
+      }
+      store_pii(doc_response)
+      user_session['idv/doc_auth']['api_upload'] = upload_info
+      success_json('Uploaded images')
+    end
+
+    def store_pii(doc_response)
+      user_session['idv/doc_auth'][:pii_from_doc] = doc_response.pii_from_doc.merge(
+        uuid: current_user.uuid,
+        phone: current_user.phone_configurations.take&.phone,
+      )
+    end
+
+    def validate_request(request)
+      steps = %i[check_content_type check_image_fields]
+      steps.each do |step|
+        err = method(step).call(request)
+        return error_json(err) if err
+      end
+      nil
+    end
+
+    def check_content_type(request)
+      "Invalid content type #{request.content_type}" if request.content_type != 'application/json'
+    end
```

Comment:
One thing that's occurring to me is that, if this endpoint receives JSON, what type of value are we expecting for each of `front_image`, `back_image`, and `selfie_image`? Would they all need to be data URLs? In the current server-rendered implementation, an image captured from Acuant would be a data URL, but if the user would upload them directly, they'd be sent as regular file form data. Submitted as `multipart/form-data`, the server seems to be able to handle the mix of types, but if we're sending pure JSON now, I assume we'd need to normalize this to the encoded string form. Or would it be better that this endpoint receives form data?

Reply:
Also should we 415 for this? https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/415

Reply:
> One thing that's occurring to me is that, if this endpoint receives JSON, what type of value are we expecting for each of `front_image`, `back_image`, and `selfie_image`? 

In a hand-off call between @jmhooper , @ironman5366 , and myself, it was determined that an initial approach here would expect to receive data URL strings as values for these properties.

Relevant follow-up: #4012